src/main/java/lotto/controller/LottoMachine.java

```
@@ -0,0 +1,87 @@
+import lotto.dto.LottoResponseDto;
+import lotto.dto.TotalPrizeResponseDto;
+import lotto.dto.WinningNumberDto;
+import lotto.mapper.AmountMapper;
+import lotto.mapper.BonusNumberMapper;
+import lotto.mapper.LottoMapper;
+import lotto.service.LottoService;
+import lotto.util.InputParser;
+import lotto.view.InputView;
+import lotto.view.OutputView;
+
+public class LottoMachine {
+
+    private final LottoService lottoService;
+
+    public LottoMachine(LottoService lottoService) {
+        this.lottoService = lottoService;
+    }
+
+    public void run() {
+        try {
+            AmountDto amount = getAmount();
+            List<LottoResponseDto> lottery = lottoService.getLottery(amount);
+
+            printLotteryByLottoResponse(lottery, amount);
+
+            WinningNumberDto winningNumber = getWinningNumber();
+            BonusNumberDto bonusNumber = getBonusNumber();
+
+            TotalPrizeResponseDto totalPrize = lottoService.getTotalPrize(lottery, winningNumber, bonusNumber);
+
+            printResult(totalPrize, amount, lottoService.getReturnRate(amount, totalPrize));
+        } finally {
+            InputView.closeConsole();
+        }
+    }
+
+    private void printResult(TotalPrizeResponseDto totalPrize, AmountDto amount, double returnRate) {
+        OutputView.printTotalPrize(totalPrize);
+        OutputView.printReturnRate(returnRate);
+    }
+
+    private void printLotteryByLottoResponse(List<LottoResponseDto> lottoResponseDto, AmountDto amount) {
+        OutputView.printLottery(AmountMapper.toEntity(amount).getCountByAmount(), lottoResponseDto);
+    }
+
+    private AmountDto getAmount() {
+        while (true) {
+            try {
+                AmountDto amountDto = InputParser.parseAmount(InputView.getAmount());
+                AmountMapper.toEntity(amountDto);
+                return amountDto;
+            } catch (IllegalArgumentException e) {
+                System.out.println(e.getMessage());
+            }
+        }
```

반복되는 구조가 많이 보이시지 않나요?!
분리의 신호로 보입니다! DRY 원칙에 대해 알아보면 좋을 것 같아요!

---

그리고 Controller에서 InputParser로 Dto 변환을 처리하고 있어요!
마치 Servlet을 보는 것 같네요 👍 

Spring Web MVC의 handling 하는 과정을 생각하면 조금 더 나은 구조가 될 것 같습니다 !

---

Mapper도 정말 필요할까요? 네이밍이  toEntity()라고 되어있는데 Entity 개념이 어떻게 나오셨는지...
영속을 관리하는게 있는지? 고민해보시면 좋겠습니다.

또 여기서는 toEntity()를 호출만 하고 있는데.. 혹시 fail fast를 고려하신 걸까요??

---

네 맞습니다. 예외 처리가 도메인 생성단에서 진행되고 있기 때문에 DTO로만 생성할 경우 도메인 내부 검증을 처리하는 부분이 아닌 경우 예외를 캐치해서 재입력을 받는 부분을 처리할 수 없었습니다. 그래서 해당 부분과 같이 코드를 작성한 것 같습니다!